evalml/tests/automl_tests/test_automl.py

```diff
@@ -5363,3 +5363,83 @@ def test_automl_respects_iterative_pipeline_order(X_y_binary, AutoMLTestEnv):
         searched_model_families,
         key=lambda model_family: _ESTIMATOR_FAMILY_ORDER.index(model_family),
     )
+
+
+def test_get_ensembler_input_pipelines(X_y_binary, AutoMLTestEnv):
+    X, y = X_y_binary
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_iterations=_get_first_stacked_classifier_no(),
+        objective="Log Loss Binary",
+        ensembling=True,
+    )
+
+    score_side_effect = [
+        {"Log Loss Binary": score}
+        for score in np.arange(
+            0, -1 * automl.max_iterations * automl.data_splitter.get_n_splits(), -0.1
+        )
+    ]  # Decreases with each call
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+
+    best_pipeline_ids = [
+        pipeline["id"]
+        for pipeline in list(automl._automl_algorithm._best_pipeline_info.values())
+    ]
+    best_pipeline_ids.sort()
+
+    input_pipeline_ids = automl.get_ensembler_input_pipelines(
+        _get_first_stacked_classifier_no() - 1
+    )
+    input_pipeline_ids.sort()
+
+    assert best_pipeline_ids == input_pipeline_ids
+
+    two_stacking_batches = 1 + 2 * (len(get_estimators(ProblemTypes.BINARY)) + 1)
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=two_stacking_batches,
+        objective="Log Loss Binary",
+        ensembling=True,
+    )
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+    pipeline_names = automl.rankings["pipeline_name"]
+    assert pipeline_names.str.contains("Ensemble").any()
```

Comment:
Is this assert carried over from a different ensemble test? I noticed that we don't test it above, and not exactly what we're testing here so we could remove it?

Reply:
Yep, removed this line.